sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionals.scala

```diff
@@ -296,9 +295,7 @@ case class CaseKeyWhen(key: Expression, branches: Seq[Expression]) extends CaseW
   }
 
   private def equalNullSafe(l: Any, r: Any) = {
```

Comment:
We should at least consider renaming this function. Null safe equals in sql typically returns `true` for `null <=> null`

http://dev.mysql.com/doc/refman/5.0/en/comparison-operators.html#operator_equal-to